src/toGamut.js

```diff
@@ -46,26 +81,59 @@ export default function toGamut (color, { method = defaults.gamut_mapping, space
 		}
 
 		if (method !== "clip" && !inGamut(color, space)) {
+
+			if (Object.prototype.hasOwnProperty.call(GMAPPRESET, method)) {
+				({method, jnd, deltaEMethod, blackWhiteClamp} = GMAPPRESET[method]);
+			}
+
+			// Get the correct delta E method
+			let de = deltaE2000;
+			if (deltaEMethod !== "") {
+				for (let m in deltaEMethods) {
+					if ("deltae" + deltaEMethod.toLowerCase() === m.toLowerCase()) {
+						de = deltaEMethods[m];
+						break;
+					}
+				}
+			}
+
 			let clipped = toGamut(clone(spaceColor), { method: "clip", space });
-			if (deltaE2000(color, clipped) > 2) {
+			if (de(color, clipped) > jnd) {
+
+				// Clamp to SDR white and black if required
+				if (Object.keys(blackWhiteClamp).length > 0) {
```

Comment:
It looks like all three `channel`, `max`, and `min` are required -- should this `if` check for them explicitly?

Reply:
You mean check for a length of 3? That seemed reasonable.

Reply:
Yeah, I was thinking either that or explicitly checking for the existence of `blackWhiteClamp.channel`, `blackWhiteClamp.max`, and `blackWhiteClamp.min`.

Reply:
I'm not sure we want to waste the time for explicit checks. For now, I'll check for 3 unless others are adamant.